repair/row_level.cc

```diff
@@ -1376,13 +1376,17 @@ class repair_meta {
 
 public:
     // Must run inside a seastar thread
-    void flush_rows_in_working_row_buf(locator::effective_replication_map_ptr erm, bool small_table_optimization) {
+    void flush_rows_in_working_row_buf(std::optional<small_table_optimization_params> small_table_optimization) {
         if (_dirty_on_master) {
             _dirty_on_master = is_dirty_on_master::no;
```

Comment:
> repair: do not pass erm to flush_rows_in_working_row_buf when unneces…
> …sary
> 
> When small_table_optimization isn't enabled, flush_rows_in_working_row_buf
> does not access erm. Add small_table_optimization_params containing erm and
> pass it only when small_table_optimization is enabled.
> 
> This is safe as erm is kept by shard_repair_task_impl.

It's unfortunate to have a function behave so differently with different parameters but I understand it's going to be a major effort to unravel it, so okay.

Reply:
Outdated